### PR TITLE
Allow star to appear in namelists

### LINKED ISSUE #1103: NLCOMP has an error when namelist contain *
CAM namelists can have fields which contain *.  When they do, NLCOMP exits with the following error. We are using cime5.2.0-alpha.22.

```
2017-02-02` 13:55:16: Exception during namelist operations:
ERROR: Incorrect usage of multiplication in token ''O3_Prod = NO_HO2 + CH3O2_NO + PO2_NO + CH3CO3_NO + C2H5O2_NO + .92*ISOPAO2_NO + .92*ISOPBO2_NO + MACRO2_NOa + MCO3_NO + C3H7O2_NO + RO2_NO + XO2_NO + .9*TOLO2_NO &''
Traceback (most recent call last):
  File "/glade/u/home/fvitt/camdev/chem_cam5_4_106/cime/scripts/Tools/../../utils/python/CIME/case_cmpgen_namelists.py", line 123, in case_cmpgen_namelists
    success, output = _do_full_nl_comp(case, test_name, compare_name, baseline_root)
  File "/glade/u/home/fvitt/camdev/chem_cam5_4_106/cime/scripts/Tools/../../utils/python/CIME/case_cmpgen_namelists.py", line 44, in _do_full_nl_comp
    if is_namelist_file(item):
  File "/glade/u/home/fvitt/camdev/chem_cam5_4_106/cime/scripts/Tools/../../utils/python/CIME/compare_namelists.py", line 540, in is_namelist_file
    assert "does not appear to be a namelist file" in str(e), str(e)
AssertionError: ERROR: Incorrect usage of multiplication in token ''O3_Prod = NO_HO2 + CH3O2_NO + PO2_NO + CH3CO3_NO + C2H5O2_NO + .92*ISOPAO2_NO + .92*ISOPBO2_NO + MACRO2_NOa + MCO3_NO + C3H7O2_NO + RO2_NO + XO2_NO + .9*TOLO2_NO &'`

### LINKED ISSUE #1115: Invalid literal error doing namelist compare
I get the following error message when doing certain namelist comparisons. 

    2017-02-06 21:02:09: Exception during namelist operations:
    invalid literal for int() with base 10: "'pom_a1 = 1.4"
    Traceback (most recent call last):
      File "/glade/u/home/fischer/code/cesm2_0_alpha06c/cime/scripts/Tools/../../utils/python/CIME/case_cmpgen_namelists.py", line 123, in case_cmpgen_namelists
        success, output = _do_full_nl_comp(case, test_name, compare_name, baseline_root)
      File "/glade/u/home/fischer/code/cesm2_0_alpha06c/cime/scripts/Tools/../../utils/python/CIME/case_cmpgen_namelists.py", line 44, in _do_full_nl_comp
        if is_namelist_file(item):
      File "/glade/u/home/fischer/code/cesm2_0_alpha06c/cime/scripts/Tools/../../utils/python/CIME/compare_namelists.py", line 538, in is_namelist_file
        compare_namelist_files(file_path, file_path)
      File "/glade/u/home/fischer/code/cesm2_0_alpha06c/cime/scripts/Tools/../../utils/python/CIME/compare_namelists.py", line 528, in compare_namelist_files
        gold_namelists = _parse_namelists(open(gold_file, "r").readlines(), gold_file)
      File "/glade/u/home/fischer/code/cesm2_0_alpha06c/cime/scripts/Tools/../../utils/python/CIME/compare_namelists.py", line 239, in _parse_namelists
        real_value = _interpret_value(value_str, filename)
      File "/glade/u/home/fischer/code/cesm2_0_alpha06c/cime/scripts/Tools/../../utils/python/CIME/compare_namelists.py", line 78, in _interpret_value
        new_tokens.extend([sub_tokens[1]] * int(sub_tokens[0]))
    ValueError: invalid literal for int() with base 10: "'pom_a1 = 1.4" 
 

This is the namelist that's causing the error.

&fire_emis_nl
  fire_emis_factors_file = '/glade/p/cesmdata/cseg/inputdata/lnd/clm2/firedata/fire_emis_factors_c140116.nc'
  fire_emis_specifier = 'bc_a1 = BC', 'pom_a1 = 1.4*OC', 'SO2 = SO2'
/
&megan_emis_nl
  megan_factors_file = '/glade/p/cesmdata/cseg/inputdata/atm/cam/chem/trop_mozart/emis/megan21_emis_factors_78pft_c20161108.nc'
  megan_specifier = 'ISOP = isoprene',
      'C10H16 = pinene_a + carene_3 + thujene_a', 'CH3OH = methanol',
      'C2H5OH = ethanol', 'CH2O = formaldehyde', 'CH3CHO = acetaldehyde',
      'CH3COOH = acetic_acid', 'CH3COCH3 = acetone'
/

### PR DESCRIPTION
**Pull request by jgfouca**
Without assuming it has to mean a multiplier

Add unit test too.

Test suite: doctest for compare_namelists.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #1115 
Fixes #1103 

User interface changes?: None

Code review: @jedwards4b

### CODE REVIEW

File: utils/python/CIME/compare_namelists.py [hunk 1]
```diff
@@ -71,11 +71,15 @@ def _interpret_value(value_str, filename):
...
-                new_tokens.extend([sub_tokens[1]] * int(sub_tokens[0]))
+                try:
+                    # the following ensure that the following to namelist settings trigger a match
+                    # nmlvalue = 1,1,1 versus nmlvalue = 3*1
+                    sub_tokens = [item.strip() for item in token.split("*")]
+                    expect(len(sub_tokens) == 2, "Incorrect usage of multiplication in token '%s'" % token)
+                    new_tokens.extend([sub_tokens[1]] * int(sub_tokens[0]))
+                except:
+                    # User probably did not intend to use the * operator as a namelist multiplier
+                    new_tokens.append(token)
```
Comment (gold2718) [inline utils/python/CIME/compare_namelists.py]: I want to make sure you are getting the right answers for the right reason.
For example, in the fire_emis example below, for the token, `'pom_a1 = 1.4*OC'`, you end up with
`sub_tokens[0] == 'pom_a1 = 1.4'` and `int(sub_tokens[0])` throws an error.
Is that what you intended? If so, please add a comment before that line so the intent is clear.

Comment (jgfouca) [inline utils/python/CIME/compare_namelists.py]: @gold2718 yes, this is intended and why I put the mutliplier code within a try block. Note that if there's any exception, it leaves the token as-is and there is a comment to this affect in the except block.

Comment (gold2718) [inline utils/python/CIME/compare_namelists.py]: That's what I figured but why the different processing for the two different expected "error" types?
Does the expect error message get printed out when inside a try? If not, why have it?
```
if len(sub_tokens) != 2:
  raise Error
```
gets the message across of what you are trying to accomplish. On the other hand, if you like the expect syntax, 
`expect(isinstance( <var>, ( int, long ) ), "Repeat count should be an integer")`
Finally, there is the null-value repeat count:
`6*,`
means six null values but your current code does not handle that type of situation.
BTW, do you have a test for null values (a null value is recognized as a comma with no preceding non-whitespace characters)?

Comment (jgfouca) [inline utils/python/CIME/compare_namelists.py]: @gold2718 I think I was just being a little lazy and using the pre-existing code to the greatest extent. I was not aware of null values.

Comment (gold2718) [inline utils/python/CIME/compare_namelists.py]: Yeah, Fortran (nee FORTRAN) is full of nooks and crannies like that. Of course, I wrote my first FORTRAN program in 1974 so I'm used to it by now :)

Comment (jgfouca) [inline utils/python/CIME/compare_namelists.py]: @gold2718 can you give a me quick example of what 6*, is supposed to look like?

**Comment by gold2718** [comment]
These two should be equivalent
```
foobar = 3, , , , , , ,9
foobar = 3,6*,9
```
What were they thinking?

**Comment by jgfouca** [comment]
@gold2718 I'm going to move that to a different issue since I think CESG wants this fix in now.

**Comment by jgfouca** [comment]
@jedwards4b please merge if you approve.

**Pull request merged**